libshared/FluidExpression.ml

```diff
@@ -457,9 +457,7 @@ let rec updateVariableUses (oldVarName : string) ~(f : t -> t) (ast : t) : t =
   | EVariable (_, varName) ->
       if varName = oldVarName then f ast else ast
   | ELet (id, lhs, rhs, body) ->
-      if oldVarName = lhs (* if variable name is rebound *)
-      then ast
-      else ELet (id, lhs, u rhs, u body)
+      ELet (id, lhs, u rhs, u body)
```

Comment:
For reference, in `ELet`, `lhs` is `x`, `rhs` is `a + b`, `body` is `z + 4`

```
let x = a + b
z + 4
```

So if `oldVarName = lhs` is true then `oldVarName` is rebound in `body` and so there should not be a rename, in which case the old code was correct. But, even if `oldVarName = lhs`, that doesn't apply in `rhs`, in which case the old code is wrong and the new code is correct.

So I think the correct code is probably:

```
if oldVarName = lhs
then ELet(id, lhs, u rhs, body)
else ELet(id, lhs, u rhs, u body)
```